### PR TITLE
WAI-946 Part 2: Add migration to split out legacy reports from map overlay table

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
WAI-946

### CODE REVIEW

File: packages/database/src/migrations/20210924062930-SplitMapOverlayLegacyReportsOut-modifies-schema.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  await db.runSql(`
+    UPDATE map_overlay
+    SET report_code = id || '_map'
+    WHERE legacy = true;
+  `);
+  await db.runSql(`
+    INSERT INTO legacy_report (id, code, data_builder, data_builder_config, data_services)
+    FROM (
+      SELECT generate_object_id(), id || '_map', "measureBuilder", "measureBuilderConfig", data_services
+      FROM map_overlay
```
Comment (EMcQ-BES) [inline packages/database/src/migrations/20210924062930-SplitMapOverlayLegacyReportsOut-modifies-schema.js:32]:
```suggestion
      FROM map_overlay
      WHERE legacy = true
```

**Review by EMcQ-BES — CHANGES_REQUESTED** [review]
One small change, otherwise migration looks good from a read through.

Obviously this is going to break everything until we actually write the supporting code.

File: packages/database/src/migrations/20210924062930-SplitMapOverlayLegacyReportsOut-modifies-schema.js (hunk 1 above)
Comment (edmofro) [reply packages/database/src/migrations/20210924062930-SplitMapOverlayLegacyReportsOut-modifies-schema.js:32]: Good catch

**Comment by edmofro** [comment]
Absolutely, I'm hoping you can pick that part up today?

**Comment by EMcQ-BES** [comment]
> Absolutely, I'm hoping you can pick that part up today?

Yep, that's the plan!

**Pull request merged**